src/macros.jl

```diff
@@ -1,25 +1,40 @@
-macro aimport(mod)
-    if isdefined(mod)
-        quote
-            import $mod
+using MacroTools
+
+function parse_deps(ex)
+    m = @match ex begin
+        (x_ <: y_) => (:deps, x, y)
+        x_ => (:no_deps, x)
+    end
+    if m[1] == :deps
+        deps = @match m[3] begin
+            [x__] => (x)
+            x_ => ([x])
         end
     else
-        esc(quote
-            arequire(string($(QuoteNode(mod))))
-            import $mod
-        end)
+        deps = []
     end
+    (m[2], deps)
 end
 
-macro ausing(mod)
-    if isdefined(mod)
+
+function auto_do(mod, action)
+    modname, deps = parse_deps(mod)
+    if isdefined(modname)
         quote
-            using $mod
+            $(Expr(action, modname))
         end
     else
         esc(quote
-            arequire(string($(QuoteNode(mod))))
-            using $mod
+            arequire(string($(QuoteNode(modname))), depends_on=map(string, $deps))
```

Comment:
Since it's in an `esc`, I think `arequire` should be `Autoreload.arequire`, otherwise `using Autoreload: @ausing` doesn't work. It works nicely so far otherwise.